content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1107 @@
+```
+
+```python
+# Run the below cell to kick off an experiment
+!det e create determined_files/const-distributed.yaml determined_files/
+```
+
+```
+Preparing files to send to master... 237.5KB and 36 files
+Created experiment 77
+```
+
+## Launching a Distributed Hyperparameter Search Experiment
+
+To implement an automatic hyperparameter tuning experiment, we need to define the hyperparameter space, e.g., by listing the decisions that may impact model performance. We can specify a range of possible values in the experiment configuration for each hyperparameter in the search space.
+
+View the `x.yaml` file that defines a hyperparameter search where we find the model architecture that achieves the best performance on the dataset.
+
+```yaml
+name: xview_frxnn_search
+workspace: Andrew
+project: Xview FasterRCNN
+profiling:
+ enabled: true
+ begin_on_batch: 0
+ end_after_batch: null
+hyperparameters:
+    lr: 0.01
+    momentum: 0.9
+    global_batch_size: 128
+    weight_decay: 1.0e-4
+    gamma: 0.1
+    warmup: linear
+    warmup_iters: 200
+    warmup_ratio: 0.001
+    step1: 18032 # 14 epochs: 14*1288 == 18,032
+    step2: 19320 # 15 epochs: 15*1288 == 19,320
+    model:
+      type: categorical
+      vals: ['fasterrcnn_resnet50_fpn','fcos_resnet50_fpn', 'ssd300_vgg16','ssdlite320_mobilenet_v3_large','resnet152_fasterrcnn_model','efficientnet_b4_fasterrcnn_model','convnext_large_fasterrcnn_model','convnext_small_fasterrcnn_model']
+
+    # Dataset
+    dataset_file: coco
+    backend: aws # specifiy the backend you want to use.  one of: gcs, aws, fake, local
+    data_dir: determined-ai-coco-dataset # bucket name if using gcs or aws, otherwise directory to dataset
+    masks: false
+    num_workers: 4
+
+    device: cuda
+environment:
+    environment_variables:                                                                          
+        - NCCL_DEBUG=INFO                                                                           
+        # You may need to modify this to match your network configuration.                          
+        - NCCL_SOCKET_IFNAME=ens,eth,ib
+bind_mounts:
+    - host_path: /tmp
+      container_path: /data
+      read_only: false
+scheduling_unit: 400
+# scheduling_unit: 40
+min_validation_period:
+    batches: 1288 # For Real training
+searcher:
+  name: grid
+  metric: mAP
+  smaller_is_better: false
+  max_length:
+    batches: 51520 # 50*1288 == 51520# Real Training
+
+records_per_epoch: 1288
+resources:
+    # slots_per_trial: 16
+    slots_per_trial: 8
+    shm_size: 2000000000
+max_restarts: 0
+
+entrypoint: python3 -m determined.launch.torch_distributed --trial model_def:ObjectDetectionTrial
+```
+
+```python
+# Run the below cell to run a hyperparameter search experiment
+!det e create determined_files/const-distributed-search.yaml determined_files/
+```
+
+```
+Preparing files to send to master... 312.2KB and 40 files
+Created experiment 79
+```
+
+## Load Checkpoint of Trained Experiment.
```

Comment:
Remove the period at the end. Change to sentence case:
Load checkpoint of trained experiment